### PR TITLE
suite for unit tests added

### PR DESCRIPTION
**Pull request by dynasource**
Test suite is added for github bot:
- phpunit.xml + bootstrap + TestCase.php
- mock files for github api
- tests for components/Github.php
- tests for commands/GithubController.php

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /config.local.php
 /tmp
 /logs
+/.idea/*
```
Comment (cebe) [inline .gitignore]: you should put this into your [global gitignore](https://gist.github.com/cebe/1166792#file-gitconfig-L27) file. The repo should only contain repo specific ignores.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+define('YII_ENABLE_ERROR_HANDLER', false);
+define('YII_ENV_TEST', true);
+
+$_SERVER['SCRIPT_NAME'] = '/' . __DIR__;
+$_SERVER['SCRIPT_FILENAME'] = __FILE__;
+
+if (is_dir(__DIR__ . '/../vendor/')) {
+    $vendorRoot = __DIR__ . '/../vendor'; //this extension has its own vendor folder
+} else {
+    $vendorRoot = __DIR__ . '/../../..'; //this extension is part of a project vendor folder
```
Comment (cebe) [inline tests/bootstrap.php]: this seems to be not needed for github-bot.

File: components/Github.php [hunk 3]
```diff
@@ -22,7 +23,7 @@ class Github extends Component
 	public function client()
 	{
 		// create client
-		$client = new \Github\HttpClient\CachedHttpClient();
+        $client = YII_ENV_TEST ? new CachedHttpClientMock() : new \Github\HttpClient\CachedHttpClient();
```
Comment (cebe) [inline components/Github.php]: better use DI container to create this instance and replace it in test env.

File: commands/GithubController.php [hunk 4]
```diff
@@ -54,7 +55,7 @@ public function actionRegister(array $limitRepos = [])
 				list($user, $repo) = explode('/', $urepo);
 
 				// https://developer.github.com/v3/repos/hooks/#create-a-hook
-				$api = new Repo($client);
+                $api = YII_ENV_TEST ? new RepoMock($client) : new Repo($client);
```
Comment (cebe) [inline commands/GithubController.php]: better use DI container to create this instance and replace it in test env.

**Review by cebe — CHANGES_REQUESTED** [review]

File: tests/bootstrap.php (hunk 2 above)
Comment (dynasource) [reply tests/bootstrap.php]: true

**Comment by cebe** [comment]
merged, thank you!

**Pull request closed without merging**